File: js_new/start-dashboard.js
#!/usr/bin/env node
/**
 * EDGE Dashboard Launcher
 * Starts the Next.js web interface for EDGE gateway monitoring
 */

import { spawn } from 'child_process';
import path from 'path';
import fs from 'fs';
import { fileURLToPath } from 'url';
import { dirname } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

function main() {
  const dashboardPath = path.join(__dirname, 'dashboard');

  // Check if dashboard directory exists
  if (!fs.existsSync(dashboardPath)) {
    console.error('❌ Dashboard directory not found:', dashboardPath);
    console.log('Run: cd js_new/dashboard && npm install');
    process.exit(1);
  }

  // Check if node_modules exists
  const nodeModulesPath = path.join(dashboardPath, 'node_modules');
  if (!fs.existsSync(nodeModulesPath)) {
    console.log('📦 Installing dashboard dependencies...');
    console.log('This may take a few minutes...');

    const install = spawn('npm', ['install'], {
      cwd: dashboardPath,
      stdio: 'inherit',
      shell: true,
    });

    install.on('close', (code) => {
      if (code !== 0) {
        console.error('❌ Failed to install dependencies');
        process.exit(1);
      }
      startDashboard();
    });
  } else {
    startDashboard();
  }
}

function startDashboard() {
  console.log('🚀 Starting EDGE Dashboard...');
  console.log('   Dashboard: http://localhost:3000');
  console.log('   API: http://localhost:8090');
  console.log('   WebSocket: ws://localhost:8000');
  console.log('   Press Ctrl+C to stop');
  console.log('');

  const dashboardPath = path.join(__dirname, 'dashboard');

  // Start Next.js development server
  const dashboard = spawn('npm', ['run', 'dev'], {
    cwd: dashboardPath,
    stdio: 'inherit',
    shell: true,
  });

  // Handle Ctrl+C
  process.on('SIGINT', () => {
    console.log('\n🛑 Stopping dashboard...');
    dashboard.kill('SIGINT');
    process.exit(0);
  });

  // Handle process errors
  dashboard.on('error', (error) => {
    console.error('❌ Dashboard error:', error);
    process.exit(1);
  });

  dashboard.on('close', (code) => {
    if (code !== 0 && code !== null) {
      console.error(`❌ Dashboard exited with code ${code}`);
      process.exit(code);
    }
  });
}

// Run main (check if this file is being run directly)
if (process.argv[1] && path.resolve(process.argv[1]) === __filename) {
  main();
}

export { main };

File: edge_clear_py/start_dashboard.py
#!/usr/bin/env python3
"""
EDGE Streamlit Dashboard Launcher
Запускает веб-интерфейс мониторинга для CUBE_RS EDGE устройств
"""

import sys
import os
import subprocess
from pathlib import Path

def main():
    # Находим путь к dashboard приложению
    dashboard_path = Path(__file__).parent / "web_dashboard" / "app.py"
    
    if not dashboard_path.exists():
        print(f"❌ Dashboard app не найден: {dashboard_path}")
        return 1
    
    print("🚀 Запуск CUBE_RS EDGE Dashboard...")
    print(f"   Dashboard: {dashboard_path}")
    print("   URL: http://localhost:8501")
    print("   Ctrl+C для остановки")
    
    # Запускаем Streamlit
    cmd = [
        sys.executable, "-m", "streamlit", "run", 
        str(dashboard_path),
        "--server.port", "8501",
        "--server.address", "0.0.0.0",
        "--theme.base", "dark"
    ]
    
    try:
        return subprocess.run(cmd).returncode
    except KeyboardInterrupt:
        print("\n🛑 Dashboard остановлен")
        return 0
    except Exception as e:
        print(f"❌ Ошибка запуска dashboard: {e}")
        return 1

if __name__ == "__main__":
    sys.exit(main())